python/cuml/common/doc_utils.py

```diff
@@ -0,0 +1,283 @@
+        will increase memory used for the method.',
+
+    'convert_dtype_other':
+    'convert_dtype : bool, optional (default = {})\n \
+        When set to True, the {} method will, when necessary, convert \
+        the input to the data type which was used to train the model. This \
+        will increase memory used for the method.',
+
+    'convert_dtype_single':
+    'convert_dtype : bool, optional (default = {})\n \
+            When set to True, the method will automatically \
+            convert the inputs to {}.',
+
+    'sample_weight':
+    'sample_weight : array-like (device or host) shape = (n_samples,), default={} \n\
+                The weights for each observation in X. If None, all observations  \
+                are assigned equal weight. \
+                Acceptable dense formats: CUDA array interface compliant objects like \
+                CuPy, cuDF DataFrame/Series, NumPy ndarray and Pandas \
+                DataFrame/Series.',  # noqa
+
+    'return_sparse':
+    'return_sparse : bool, optional (default = {}) \n \
+            Ignored when the model is not fit on a sparse matrix \
+            If True, the method will convert the result to a \
+            cupy.sparse.csr_matrix object. \n \
+            NOTE: Currently, there is a loss of information when converting \
+            to csr matrix (cusolver bug). Default will be switched to True \
+            once this is solved.',
+
+    'sparse_tol':
+    'sparse_tol : float, optional (default = {}) \n\
+            Ignored when return_sparse=False. \
+            If True, values in the inverse transform below this parameter \
+            are clipped to 0.',
+
+    '_custom_docstring_default':
+    '{}: {} (default = {}) \n \
+        {}',
+
+    '_custom_docstring_no_default':
+    '{}: {} \n \
+        {}'
+}
+
+_parameter_possible_values = ['name',
+                              'type',
+                              'shape',
+                              'default',
+                              'description',
+                              'accepted']
+
+_return_values_docstrings = {
+    'dense':
+    '{} : cuDF, CuPy or NumPy object depending on cuML\'s output type configuration, shape = {}\n \
+        {} \n For more information on how to configure cuML\'s output type, \
+        refer to: `Output Data Type Configuration`_.',  # noqa
+
+    'dense_sparse':
+    '{} : cuDF, CuPy or NumPy object depending on cuML\'s output type configuration, cupy.sparse for sparse output, shape = {}\n \
+        {} \n For more information on how to configure cuML\'s dense output type, \
+        refer to: `Output Data Type Configuration`_.',  # noqa
+
+    'dense_datatype':
+    'cuDF, CuPy or NumPy object depending on cuML\'s output type \
+    configuration, shape ={}',
+
+    'dense_sparse_datatype':
+    'cuDF, CuPy or NumPy object depending on cuML\'s output type \
+    configuration, shape ={}',
+
+    'custom_type':
```

Comment:
For custom_type, maybe just allow the user to write the docstrings directly instead of splicing it in here? Like allow the user to set return type as None then write a custom returns block

Reply:
These are not really used (for now at least), so removed them until we have a use case that would show what's the best approach